chef/automate
# Adding compliance page filters to the URL

### :nut_and_bolt: Description: What code changed, and why?
Updating the compliance reports page to read and write all filters to the URL. This allows users to save and share their filters to be loaded later in a fresh Automate webpage. 

Below are the filters that are added to the URL
* End Date
* Time Interval index (10 days, month, 3 months, year)
* All search bar filters
-- Chef Server
-- Chef Tags
-- Control
-- Environment
-- InSpec Version
-- Node Name
-- Organization
-- Platform
-- Policy Group
-- Policy Name
-- Profile
-- Recipe
-- Role

One complexity is that the profiles, nodes, and controls are added to the URL with their IDs. On refresh, the filter bar (shown below) will display the IDs and not the name of the object.

Below shows the filter bar when a profile and node is first added. The names of the objects are used. 
<img width="1427" alt="Screen Shot 2019-09-04 at 3 38 32 PM" src="https://user-images.githubusercontent.com/1679247/64298344-133f4380-cf2a-11e9-99b4-1d6aa1c10045.png">
Below shows the filter bar when the page is refreshed or shared with another user. The IDs of the objects are used. 
<img width="1419" alt="Screen Shot 2019-09-04 at 3 39 58 PM" src="https://user-images.githubusercontent.com/1679247/64298460-77620780-cf2a-11e9-99bf-f6ff4a733788.png">

### :chains: Related Resources
#649

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-ui && start_all_services`
1. Add compliance nodes with `chef_load_compliance_scans -N10 -M100 -T10 -D10`
1. Open the compliance report page https://a2-dev.test/compliance/reports/overview
1. Add filters with the search bar. 
1. Refresh the page or copy and paste the URL into a different tab. Ensure that the same filters are selected. 
1. Change the end date (<img width="111" alt="Screen Shot 2019-09-04 at 3 45 46 PM" src="https://user-images.githubusercontent.com/1679247/64298625-17b82c00-cf2b-11e9-978f-6c6a63e9555e.png">). Refresh the page. Ensure that the same end date is selected. 
1. Change the date interval with the Node Status Over Time (<img width="190" alt="Screen Shot 2019-09-04 at 3 47 14 PM" src="https://user-images.githubusercontent.com/1679247/64298700-54842300-cf2b-11e9-9fc0-21df5e521a2f.png">). 
1. Refresh the page and ensure the same date interval is selected. 
1. Select the "Nodes" section (<img width="318" alt="Screen Shot 2019-09-04 at 3 48 37 PM" src="https://user-images.githubusercontent.com/1679247/64298754-7da4b380-cf2b-11e9-883e-305bc5aad166.png">). 
1. Add a node to the filter with the plus button. <img width="181" alt="Screen Shot 2019-09-04 at 3 49 32 PM" src="https://user-images.githubusercontent.com/1679247/64298808-a036cc80-cf2b-11e9-8fbd-267ea8631718.png">.
1. Ensure the node was added to the filter bar and the URL. Refresh the page and ensure the node is still in the filters. 
1. Select the Profiles section and add a profile to the filter. 
1. Ensure the profile was added to the filter bar and the URL. Refresh the page and ensure the profile is still in the filters. 
1. With some filters in the filter bar, press the "Clear All" button and ensure all the search bar filters are removed. The end date and date interval filters will not be removed. 
1. Lastly test the copy URL button <img width="59" alt="Screen Shot 2019-09-04 at 3 52 48 PM" src="https://user-images.githubusercontent.com/1679247/64298936-1e936e80-cf2c-11e9-8782-f386622b86cc.png">
1. Ensure all filters are copied. 

### :white_check_mark: Checklist
